addons/web/static/src/core/utils/hooks.js

```diff
@@ -336,3 +336,85 @@ export function useRefListener(ref, ...listener) {
         () => [ref.el]
     );
 }
+
+/**
+ * This is used on text inputs to automatically resize the input based on its
+ * content each time it is updated. It takes the reference to the input as
+ * parameter. Do note that it may introduce mild performance issues since it will
+ * force a reflow of the layout each time the input is updated.
+ *
+ * @param {Ref} ref
+ */
+export function useAutoresizeInput(ref) {
+    useEffect(() => {
+        if (ref.el) {
+            autoresize();
+        }
+    });
+    useRefListener(ref, "input", autoresize);
+    function autoresize() {
+        // This mesures the maximum width of the input which can get from the flex layout.
+        ref.el.style.width = "100%";
+        const maxWidth = ref.el.clientWidth;
+        // Minimum width of the input
+        ref.el.style.width = "10px";
+        if (ref.el.value === "" && ref.el.placeholder !== "") {
+            ref.el.style.width = "auto";
+            return;
+        }
+        if (ref.el.scrollWidth + 5 > maxWidth) {
+            ref.el.style.width = "100%";
+            return;
+        }
+        ref.el.style.width = ref.el.scrollWidth + 5 + "px";
+    }
+}
+
+/**
+ * This is used on textareas to automatically resize the height based on its
+ * content each time it is updated. It takes the reference to the textarea,
+ * the reference to its parent div and the minimumHeight of this div as
+ * parameters. Do note that it may introduce mild performance issues since it will
+ * force a reflow of the layout each time the textarea is updated.
+ *
+ * @param {Ref} textareaRef
+ * @param {Ref} parentDivRef
+ * @param {Number} minimumHeight
+ */
+export function useAutoresizeTextarea(textareaRef, parentDivRef, minimumHeight = 0) {
```

Comment:
no need for the parent div ref, you can just use the parentElement key on the text area

Reply:
ok, in my opinion, here is a list of changes that you should do:
- move all autoresize code to `src/core/utils/autoresize`, having a giant `hook` file is not helpful
- only export one hook, having 2 different hooks that perform essentially the same job is slightly annoying. Here is what its api should be:
```js
export function useAutoResize(ref, options = {}) { 
  // for now, only option is minimumHeight
  // ref is either an input or a textarea
  ...
}
```
Here is an outline of what your file should roughly look like:
```js
function autoResizeInput(inputEl) {
  // do something
}

function autoResizeTextarea(textareaEl, options) {
 // ...
}

export function useAutoResize(ref, options = {}) { 
    useEffect(() => {
        if (ref.el) {
            resize();
        }
    });
    useRefListener(ref, "input", () => {
      if (ref.el instanceof InputElement) {
        autoResizeInput(ref.el);
      } else if (ref.el instanceof TextAreaElement) {
         autoResizeTextArea(ref.el, options)
      }
    });
}
```